src/hooks/usePersistSelection.ts

```diff
@@ -0,0 +1,48 @@
+import isObject from 'lodash/isObject';
+import {useEffect, useState} from 'react';
+import CONST from '@src/CONST';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+import useMobileSelectionMode from './useMobileSelectionMode';
+import useResponsiveLayout from './useResponsiveLayout';
+
+type FilterOption<T> = Record<string | number, T>;
+
+function isPendingValues<T>(option?: T) {
+    if (isEmptyObject(option) || !isObject(option)) {
+        return true;
+    }
+    return 'pendingAction' in option && option?.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE;
+}
+
+function usePersistSelection<T>(options: FilterOption<T> | undefined, filterFn: (option?: T) => boolean = isPendingValues) {
```

Comment:
Can we use an array instead of records? Since records values would typically always have a value of `true`.

The new arrays values should be the current object keys